aptos-move/framework/aptos-experimental/sources/trading/market/market.move

```diff
@@ -98,7 +107,11 @@ module aptos_experimental::market {
             // Incremental fill id for matched orders
             next_fill_id: u64,
             config: MarketConfig,
-            order_book: OrderBook<M>
+            order_book: OrderBook<M>,
+            /// Pre cancellation tracker for the market, it is wrapped inside a table
+            /// as otherwise any insertion/deletion from the tracker would cause conflict
+            /// with the order book.
+            pre_cancellation_tracker: Table<u8, PreCancellationTracker>,
```

Comment:
looking at it, I think it's worth making this a bit cleaner. but let's land yours, and I'll send a PR separately to make this nicer.